From 3deadbfcdf776eb0c994bb4d719e601160943bfa Mon Sep 17 00:00:00 2001
From: Fabian Vogt <fabian@ritter-vogt.de>
Date: Wed, 28 Aug 2019 15:09:49 +0200
Subject: [PATCH] Set GTK_BACKEND=x11 in a wayland session

Works around missing window decorations and broken config file reading
---
 startkde/startplasma-waylandsession.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/startkde/startplasma-waylandsession.cpp b/startkde/startplasma-waylandsession.cpp
index 87c71c6b3..5d0b3f65f 100644
--- a/startkde/startplasma-waylandsession.cpp
+++ b/startkde/startplasma-waylandsession.cpp
@@ -45,6 +45,11 @@ int main(int /*argc*/, char** /*argv*/)
     out << "startplasma-waylandsession: Starting up...";
 
     if (qEnvironmentVariableIsSet("DISPLAY")) {
+        // GTK3 uses the wayland backend by default, but its implementation is not correct
+        // enough to work well here. Window decorations are missing, for instance.
+        if (!qEnvironmentVariableIsSet("GDK_BACKEND")) {
+            qputenv("GDK_BACKEND", "x11");
+        }
         setupX11();
     } else {
         qWarning() << "running kwin without Xwayland support";
-- 
2.22.0

